server_test.cpp

```diff
@@ -0,0 +1,29 @@
+#include "gtest/gtest.h"
+#include "server.hpp"
+
+class ServerTest : public ::testing::Test {
+protected:
+	bool checkAccept(boost::system::error_code ec) {
+		http::server::server server_("127.0.0.1", "8080");
+		return server_.checkServer(ec);
+	}
+
+};
+
+TEST_F(ServerTest, ServerStatus) {
+	boost::system::error_code ec = boost::system::errc::make_error_code(boost::system::errc::success);
```

Comment:
Adding custom error codes without actually capturing them from our program accomplishes nothing. Please catch and throw the error from server constructor. Do not add custom errors like this.